plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/preferences/OCBinary.java

```diff
@@ -139,4 +146,26 @@ public boolean isCompatibleForPublishing(IProgressMonitor monitor) {
 	public boolean isCompatibleForPublishing(IConnection connection, IProgressMonitor monitor) {
 	    return new OCBinaryVersionValidator(getLocation(connection)).isCompatibleForPublishing(monitor);
 	}
+	
+	/**
+	 * Compute the error message for the OCBinary state and path.
+	 * 
+	 * @param valid if the oc binary is valid or not
+	 * @param location the location of the oc binary
+	 * @return the error message (may be null)
+	 */
+	public IStatus getOCBinaryStatus(IConnection connection, IProgressMonitor monitor) {
```

Comment:
I see #getOCBInaryStatus being moved from the server page to OCBinary. Which seems like a great change  to me (in terms of encapsulation and so that it can be reused). 
1) Can now OCBinary#getLocation get private?
2) Shouldn't we now replace all #isCompatibleForPublishing by #getOCBinaryStatus?
3) Can't we rename #getOCBinaryStatus to #isValid ? (the name #getOCBinaryStatus is obviously repeating an obvious point: it's about the status of the binary. But this is obvious since we're in the OCBinary class anyhow. Then #getStatus seems not to nail the purpose since it only checks binary location and version, not some binary state.

Reply:
1. `public String getLocation()` is already marked `@Deprecated`. `public String getLocation(IConnection connection)` is used in `AbstractOpenShiftCliHandler` and `OCBinaryOperation`. So i don't think that it's worth making it private now. Though need to look deeper and not as part of this pr.
2.  +1. Actually, all `isCompatibleForPublishing` are not used anymore anywhere. So they can be even removed now.
3. +1

Reply:
@adietish read once again 3rd point. Maybe i didn't understand you completely,  but i think `getStatus` is the best option. `isValid` doesn't match well, because smth starting from "is" must return binary: true\false